includes/controls/media.php

```diff
@@ -271,9 +271,50 @@ public function content_template() {
 			<# } #>
 			<input type="hidden" data-setting="{{ data.name }}"/>
 		</div>
+
+		<# if ( data.sizes_supported ) { #>
+
+		<div class="elementor-control-content elementor-control-media-size">
+			<div class="elementor-control-field elementor-control elementor-control-type-select">
+				<label for="elementor-control-default" class="elementor-control-title"><?php echo esc_html__( 'Image Size', 'elementor' ); ?></label>
+				<div class="elementor-control-input-wrapper media-size-input-wrapper elementor-control-unit-5">
+				<select data-setting="size" class="elementor-control elementor-control-type-select elementor-control-input-wrapper elementor-control-media-display-size select">
+				<?php
+				$sizes = $this->get_image_sizes_options();
+				foreach ( $sizes as $key => $value ) {
+					?>
+						<option value="<?php echo esc_attr( $key ); ?>" 
+						{{ data.controlValue.size && data.controlValue.size == '<?php echo esc_attr( $key ); ?>' ? 'selected' : '' }}
+						{{ ! data.controlValue.size && '<?php echo esc_attr( $key ); ?>' == 'full' ? 'selected' : '' }}
```

Comment:
maybe extract this logic to a function above the render?

Reply:
What for? It won't be reused.

Reply:
Done